src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/executor/SaveChecker.java

```diff
@@ -30,4 +32,17 @@ public interface SaveChecker {
    * @return The <scriptName, chunkCheckPoint> map indicating where the new run should start from.
    */
   ImmutableMap<String, ChunkCheckpoint> getScriptCheckPoints(Path path);
+
+  /**
+   * Given a set of scripts, returns the names of those that have finished successfully during a
+   * previous run. The "finishedness" is inferred from the names of the files present in the target
```

Comment:
nit: completed / completeness sounds more natural

Reply:
Good call, although complete/completeness may have overloaded meanings in the context of data chunking. I rephrased it to comply more with the teachings of tech writers. :-)